doc/design/libsvm_format_column.md

```diff
@@ -0,0 +1,134 @@
+
+- The actual label value used for training would be confused if the LibSVM format column contains label value.
+    - Users may choose the label in the LibSVM format column as the training label, but we have not supported to choose partial data of a column as training label yet.
+    - Users may choose another column as the training label, which is different from the label value in the LibSVM format column.
+- If the LibSVM format column contains label value, users can easily split the label values into a separate column. For example, the SQL statement in MySQL is like:
+    ```sql
+    CREATE TABLE split_table AS 
+    SELECT 
+    SUBSTRING_INDEX(libsvm_column_with_label, ' ', 1) label, 
+    TRIM(SUBSTR(libsvm_column_with_label, LOCATE(' ', libsvm_column_with_label))) libsvm_column_without_label
+    FROM source_table;
+    ```
+    
+Therefore, we choose to only support the LibSVM format column without label value in this design. If the LibSVM format column contains label value, users can split the label values to a separate column using the SQL statement above.
+
+## Design
+
+### Add Format Field in FieldDesc
+
+Add a field named `format` in `FieldDesc`. It may be `CSV`, `LibSVM`, or other data format we would support in the future.
+
+```go
+type FieldDesc struct {
+   Name      string
+   DType     int
+   Delimiter string
+   Shape     []int
+   IsSparse  bool
+   Format    string  // indicates the data format
+   ...
+}
+```
+
+### Implicit Data Format Derivation in Feature Derivation Stage
+
+In the feature derivation stage, we can use a regular expression to check whether the column is a LibSVM format column. To avoid too much regular expression matching, this checking would be done only once when inferring the first row of the fetched samples in feature derivation stage. That is to say (in pseudo codes):
+
+```go
+func InferFeatureColumns() {
+    rows := FetchSamples()
+    rowCount := 0
+    for rows.Next() {
+        if rowCount == 0 {
+            format := inferDataFormat(rows.Value()) // Use regular expression to infer the data format
+            if format == "LibSVM" {
+                // Fill FieldDesc info when the data format is LibSVM
+                // For example: FieldDesc.Format = "LibSVM", FieldDesc.IsSparse = true, etc.
+                ...
+            }
+        }
+        rowCount ++
+    }
+}
+```
+
+### Convert the Values of the LibSVM Format Column to Sparse Tensor in Python Side
+
+After we have inferred the format of each column in the feature derivation stage, and the data format information would be used in Python code generation. In the Python side, we would first read the raw data from the database, and then convert the data from the LibSVM format column into sparse tensors for training, prediction, evaluating, and explaining.
```

Comment:
For XGBOOST，we won’t convert the data to tensor

Reply:
Thanks! Fixed.